lib/active_merchant/billing/credit_card.rb

```diff
@@ -362,7 +362,7 @@ def validate_verification_value #:nodoc:
           unless valid_card_verification_value?(verification_value, brand)
             errors << [:verification_value, "should be #{card_verification_value_length(brand)} digits"]
           end
-        elsif requires_verification_value?
+        elsif requires_verification_value? && !valid_card_verification_value?(verification_value, brand)
```

Comment:
might be a good idea to move the `valid_card_verification_value?` up a level since it's use in both branches. Something like
```ruby
unless valid_card_verification_value?(verification_value, brand)
  if verification_value?
    errors << [:verification_value, "should be #{card_verification_value_length(brand)} digits"]
  elsif requires_verification_value?
     errors << [:verification_value, 'is required']
  end
end
```